modules/custom/d_content/modules/d_content_init/src/Services/ConfigUpdate.php

```diff
@@ -27,17 +28,27 @@ class ConfigUpdate {
    */
   protected $moduleHandler;
 
+  /**
+   * Logger factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactory
+   */
+  protected $loggerFactory;
+
   /**
    * ConfigUpdate constructor.
    *
    * @param \Drupal\Core\Config\ConfigFactory $configFactory
    *   Config factory.
    * @param \Drupal\Core\Extension\ModuleHandler $moduleHandler
    *   Module handler.
+   * @param \Drupal\Core\Logger\LoggerChannelFactory $loggerFactory
+   *   Logger factory.
    */
-  public function __construct(ConfigFactory $configFactory, ModuleHandler $moduleHandler) {
+  public function __construct(ConfigFactory $configFactory, ModuleHandler $moduleHandler, LoggerChannelFactory $loggerFactory) {
     $this->configFactory = $configFactory;
     $this->moduleHandler = $moduleHandler;
+    $this->loggerFactory = $loggerFactory;
```

Comment:
a co Ty na to, żeby tutaj pobrać już `channel('d_content_init')` i przypisać do zmiennej ? nie trzeba by było potem w kodzie przy okazji każdego logowania błędu wywoływac channel :thinking:

Reply:
To jeszcze jako ciekawostka, nie musisz wstrzykiwać serwisu :-) jest calkiem przyjazny trait `LoggerChannelTrait`
który dostarcza metodę `$this->getLogger('channel_name')`. Oczywiście nie zmienia to faktu, że bym przypisał logger z konkretnym kanałem do $this->logger.

Reply:
Celna uwaga :)